test/docs/debug.test.js

```diff
@@ -67,4 +71,20 @@ describe('debug: shell', function() {
     assert.equal(true, lastLog.includes('ISODate'));
   });
 
+  it('should allow to set the `debug` option on a per-connection basis (gh-12700)', async function() {
+    // Disable global debug
+    mongoose.set('debug', false);
```

Comment:
I'd recommend creating a new mongoose instance with `const m = new mongoose.Mongoose()`  and `m.createConnection()` instead of modifying the Mongoose singleton. Modifying global state in tests often causes confusing issues.